test/test_notify_bugs.py

```diff
@@ -143,14 +143,20 @@ def test_update_metadata(env, git_gecko, git_wpt, pull_request, git_wpt_metadata
     with patch("sync.notify.bugs.components_for_wpt_paths",
                return_value={"Testing :: web-platform-tests": ["test/test.html"]}):
         with patch("sync.notify.bugs.Mach", return_value=mock_mach(None)):
-            with SyncLock.for_process(sync.process_name) as lock:
-                with sync.as_mut(lock):
-                    bug_data = bugs.for_sync(sync, results_obj)
-                    bugs.update_metadata(sync, bug_data)
+            with patch("sync.meta.Metadata.github") as mock_github:
+                with SyncLock.for_process(sync.process_name) as lock:
+                    with sync.as_mut(lock):
+                        bug_data = bugs.for_sync(sync, results_obj)
+                        bugs.update_metadata(sync, bug_data)
+
+                assert mock_github.create_pull.called
+                head = mock_github.method_calls[0].args[-1]
+
     bugs_filed = bug_data.keys()
     assert len(bugs_filed) == 1
     bug = bugs_filed[0]
```

Comment:
Is there an example of `bug_data`?
because everything seems to say that it's a dictionary with only one key.

Reply:
It's a mapping of `{bug_id: test results}` where the test results are a list of results that were included in that bug. I've added a docstring to the `for_sync` function that hopefully makes this clearer.

Reply:
ok that was just to confim. 
Another way would be:

```python
assert len(bug_data) == 1
[bug, test_results] = bug_data.items()
assert link.url == "%s/show_bug.cgi?id=%s" % (env.bz.bz_url, bug)
```

just fyi

Reply:
It would actually be `[(bug, test_results)] = bug_data.items()` which I guess implicitly asserts that the length is 1, but I'm not sure it's obviously more clear :)